### PR TITLE
Add ENABLE_OPENTELEMETRY flag 

### PR DESCRIPTION
**Pull request by SpiritZhou**
Add the ENABLE_OPENTELEMETRY flag to control the installation of OpenTelemetry during the deployment or testing process.

### Checklist

- [X] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [X] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [X] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #https://github.com/kedacore/keda/issues/5375

### CODE REVIEW

File: .github/workflows/pr-e2e.yml [hunk 1]
```diff
@@ -155,6 +155,7 @@ jobs:
           AZURE_RUN_AAD_POD_IDENTITY_TESTS: true
           AZURE_RUN_WORKLOAD_IDENTITY_TESTS: true
           GCP_RUN_IDENTITY_TESTS: true
+          ENABLE_OPENTELEMETRY : true
```
Comment (JorTurFer) [inline .github/workflows/pr-e2e.yml:158]: As this file is got from `main` branch, we can't test this right now. We have 2 options (and both are valid at least to me), 
- Adding this field (here an also in the other yaml) in other PR that we can merge before this
- Removing this field from here and the other yaml and I can add it as secret in the repo (which is dynamically loaded so it doesn't require any other PR)

I guess that to be consistent with other flags, we should add it in other split PR, but as you prefer

File: CHANGELOG.md [hunk 2]
```diff
@@ -62,7 +62,7 @@ Here is an overview of all new **experimental** features:
 
 ### Improvements
 
-- **General**: TODO ([#XXX](https://github.com/kedacore/keda/issues/XXX))
+- **General**: Introduce ENABLE_OPENTELEMETRY in deploying/testing process  ([#5375](https://github.com/kedacore/keda/issues/5375))
```
Comment (zroubalik) [inline CHANGELOG.md:65]: Nit, please move this into Other section, as it is not user facing change

**Comment by SpiritZhou** [comment]
My local testing works fine, and I added some logs for checking. Could you help to run e2e test again? @JorTurFer

File: pkg/scaling/cache/scalers_cache.go [hunk 3]
```diff
@@ -138,6 +138,7 @@ func (c *ScalersCache) GetMetricsAndActivityForScaler(ctx context.Context, index
 	}
 	startTime = time.Now()
 	metric, activity, err = ns.GetMetricsAndActivity(ctx, metricName)
+	log.Info("Get latency from GetMetricsAndActivityForScaler:", "latency", time.Since(startTime).Milliseconds(), "metricName", metricName, "err", err)
```
Comment (zroubalik) [inline pkg/scaling/cache/scalers_cache.go:141]: Why this change please?

Comment (SpiritZhou) [reply pkg/scaling/cache/scalers_cache.go:141]: I just added logs temporarily for checking purposes, but the test passed after I added logs. It seems like the latency metric is intermittent.

**Comment by SpiritZhou** [comment]
Removed debugging log in scaler_cache.go

**Review by JorTurFer — APPROVED** [review]

**Comment by JorTurFer** [comment]
Could you fix the conflict?

**Review by zroubalik — COMMENTED** [review]
We should add `ENABLE_OPENTELEMETRY = true` to GH Secrets, right?

**Comment by tomkerkhove** [comment]
This is not a secret though, so let's add it to variables to not mask the value

**Review by zroubalik — COMMENTED** [review]
LGTM, thanks!
Screenshot 2024-02-20 at 9 51 32
